docs/source/migration.mdx

```diff
@@ -47,20 +40,157 @@ graph TB;
+- In Apollo Server 3, the framework integrations automatically set up their framework's HTTP body parsing and CORS header writing for you, and you could configure that functionality in a framework-specific way via the Apollo Server API. In Apollo Server 4, it's your responsibility to set up these standard features yourself. Specifically, when using `expressMiddleware`, you should install the `body-parser` and `cors` npm packages and use them in your Express app, just like with any other JSON-based API server. Note that `startStandaloneServer` does set up these packages for you automatically, but does not allow you to configure their behavior.
+- In Apollo Server 3, some framework integrations take a `path` option that lets you configure on what URL path they listen on, with a default of `/graphql`. This feature overlapped with the router features defined by web frameworks and has been removed in Apollo Server 4. This means that to provide the same API when migrating, you'll want to explicitly mount `expressMiddleware` at `/graphql`. (Note that the AS3 batteries-included server `apollo-server` listened on all paths, not just `/graphql`.)
+
+The follow sections show how servers using `apollo-server` and `apollo-server-express` need to change to use the new API.
+
+### Migrating from `apollo-server`
+
+If you used the batteries-included `apollo-server` package in Apollo Server 3, you'll want to use the `startStandaloneServer` function in Apollo Server 4. Call `startStandaloneServer(server)` instead of `server.listen()`. Note that you cannot configure CORS settings with `startStandaloneServer`; if you want to change CORS settings from the default, use `expressMiddleware` instead.
+
+<MultiCodeBlock>
+
+```ts title="apollo-server-3-standalone.ts"
+// npm install apollo-server graphql
+import { ApolloServer } from 'apollo-server';
+import { typeDefs, resolvers } from './schema';
+
+interface MyContext {
+  token?: String;
+}
+
+async function startApolloServer() {
+  const server = new ApolloServer({
+    typeDefs,
+    resolvers,
+    context: async ({ req }) => ({ token: req.headers.token }),
+  });
+  const { url } = await server.listen(4000);
+  console.log(`🚀  Server ready at ${url}`);
+}
+```
+
+</MultiCodeBlock>
+
+<MultiCodeBlock>
+
+```ts title="apollo-server-4-standalone.ts"
+// npm install @apollo/server graphql
+import { ApolloServer } from '@apollo/server';
+import { startStandaloneServer } from '@apollo/server/standalone';
+import { typeDefs, resolvers } from './schema';
+
+interface MyContext {
+  token?: String;
+}
+
+async function startApolloServer() {
+  const server = new ApolloServer<MyContext>({ typeDefs, resolvers });
+  const { url } = await startStandaloneServer(server, {
+    context: async ({ req }) => ({ token: req.headers.token }),
+    listen: { port: 4000 },
+  });
+  console.log(`🚀  Server ready at ${url}`);
+}
+```
+
+</MultiCodeBlock>
+
+
```

Comment:
Hmm, I do like it being separated instead of in the big paragraph like I wrote it, but I think it would be better before the code? That would also address your previous comment about this being the first time people are seeing it.

Reply:
Okay, how would you feel about moving:

> Note that we start our server using `startStandaloneServer(server)` (instead of calling `server.listen()`). 

before the code snippet, then leaving this final callout:

> > The `startStandaloneServer` function doesn't enable you to configure your server's CORS setting. If you want to customize your CORS settings, use the [`expressMiddleware` function](#migrating-from-apollo-server-express) instead.

as I suggested at the end of the section?

Reply:
I like that.